cns/requestcontroller/kubecontroller/crdreconciler.go

```diff
@@ -16,44 +19,132 @@ import (
 type CrdReconciler struct {
 	KubeClient KubeClient
 	NodeName   string
+	Namespace  string
 	CNSClient  cnsclient.APIClient
 }
 
 // Reconcile is called on CRD status changes
 func (r *CrdReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	var nodeNetConfig nnc.NodeNetworkConfig
+	var (
+		nodeNetConfig nnc.NodeNetworkConfig
+		ipConfigs     []*cns.ContainerIPConfigState
+		cntxt         context.Context
+		err           error
+	)
 
 	//Get the CRD object
-	if err := r.KubeClient.Get(context.TODO(), request.NamespacedName, &nodeNetConfig); err != nil {
+	cntxt = context.TODO()
+	if err = r.KubeClient.Get(cntxt, request.NamespacedName, &nodeNetConfig); err != nil {
 		if apierrors.IsNotFound(err) {
 			logger.Printf("[cns-rc] CRD not found, ignoring %v", err)
 			return reconcile.Result{}, client.IgnoreNotFound(err)
 		} else {
 			logger.Errorf("[cns-rc] Error retrieving CRD from cache : %v", err)
+			//requeue
 			return reconcile.Result{}, err
 		}
 	}
```

Comment:
Not sure if i understood the change, Request controller will call the NC Request only to CNSClient, no?